src/shims/unix/unnamed_socket.rs

```diff
@@ -171,82 +167,143 @@ impl FileDescription for AnonSocket {
 }
 
 /// Write to AnonSocket based on the space available and return the written byte size.
+/// This is only for socketpair/pipe without O_NONBLOCK flag.
```

Comment:
Just something on top of my mind while working on this, if we were to support ``O_NONBLOCK`` flag in ``fcntl``, which has the ability of setting a file descriptor with blocking ability to non-blocking, we need to be careful of it's interaction with all the blocking operation here.